scrapers/ri/events.py

```diff
@@ -92,14 +93,21 @@ def scrape_agenda(self, chamber, url):
                 break
             except ValueError:
                 continue
-
-        event = Event(
-            name=event_desc, start_date=self._tz.localize(datetime), location_name=where
-        )
+        event_start = self._tz.localize(datetime)
+        event = Event(name=event_desc, start_date=event_start, location_name=where)
 
         event.add_document("Agenda", url, media_type="text/html", on_duplicate="ignore")
         event.add_source(url)
-        event.dedupe_key = event_desc
+
+        # Unique key for preventing triggering of duplicate item error
+        event_details_key = f"{event_desc}#{event_start}#{where}"
+
+        if event_details_key in self.event_keys:
+            self.warning(f"Duplicate event {event.dedupe_key}")
```

Comment:
Technically, this won't skip the duplicate. We'll just get a warning and then when we go to insert we'll still get the duplicate item error.
We probably want to re-do line 190 like (roughly) so:

```
events = set()
for chamber in chambers:
    for event in self.scrape_chamber(chamber):
        if event in events:
            self.warning(f"Duplicate event: {event}")
            continue
        else:
            events.add(event)
        yield event
````

Reply:
This ensures we don't send duplicate events to the insert step, but we're still notified that they're happening, which should help with potential debugging later.

Reply:
I dig it.